developers/arbitrum-deploy.md

```diff
@@ -1,121 +1,498 @@
-[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" # This loads nvm
-nvm install 16.20.0
-nvm use 16.20.0
-node --version
-npm install --global yarn
-yarn --version
+Since the contracts deployed through the factories above are already configured
+to communicate with Blobstream, you now only have to configure your node
+accordingly. First understand the different variables that will be set in the config:
+
+- **`enable`:** self explanatory, set it to true if you are using Celestia DA 😁
+- **`rpc`:** RPC endpoint for **celestia-node**
+- **`tendermint-rpc`:** a celestia-core endpoint from a full node
+(**NOTE:** only needed for a batch poster node)
+- **`eth-rpc`:** Ethereum Client RPC endpoint, only used when the node is a batch poster
+- **`namespace-id`:** namespace being used to post data to Celestia
+- **`auth-token`:** auth token for your Celestia Node
+- **`is-poster`:** is the node with Celestia DA the batch poster, set to true if so.
+- **`gas-price`:** how much to pay for gas (in uTIA)
+- **`event-channel-size`:** size of the events channel used by the batch poster
+to wait for a range of headers that contains the header for the block in which
+it posted a blob, before posting the batch to the base layer for verification
+on Blobstream X.
+- **`blobstreamx-address`:** address of the Blobstream X contract on the base chain.
+    - Note that the `SequencerInbox` contract for each chain has a constant
+    address for the `BlobstreamX` contract, thus make sure that the Blobstream X
+    address in the `SequencerInbox` being used for the templates in
+    `RollupCreator` matches the one in your config.
+
+Now enable Celestia DA in your Arbitrum chain params in
+`config/nodeConfig.json`. If you'd like to use your own namespace,
+use a custom 10 byte value or random value using
+`openssl rand -hex 10` for `namespace-id`:
+
+:::warning
+The Orbit contracts depend on
+[the existing Blobstream X deployments](#blobstream-x-contract-deployments).
+Before using these addresses, **please verify the contract addresses** on
+[the official source below](#blobstream-x-contract-deployments)
+to avoid any issues due to incorrect addresses.
+This is crucial to protect against potential misuse by copy-paste errors.
+:::
+
+```ts
+"celestia-cfg": {
+  "enable": true,
+    "rpc": "http://host.docker.internal:26658",
+    "tendermint-rpc": "http://consensus-full-mocha-4.celestia-mocha.com:26657",
+    "eth-rpc": "<YOUR_ETH_RPC>",
+    "namespace-id": "000008e5f679bf7116cb",
+    "auth-token": "<YOUR_AUTH_TOKEN>",
+    "is-poster": true,
+    "gas-price": 0.3,
+    "event-channel-size": 100,
+    "blobstreamx-address": "0xc3e209eb245Fd59c8586777b499d6A665DF3ABD2",
+    }
 ```
 
-<!-- markdownlint-enable MD013 -->
+[See the compatibility matrix in the appendix to verify you're using the right versions.](#compatibility-matrix)
 
-### Other Dependencies
+### Step 7: Pick an L2 RPC URL
```

Comment:
and do we need this step still? @Ferret-san

Reply:
Yes, but here's an version of it:

```
Step 7: Pick an L2 RPC URL
In order for the Batch Poster to subscribe to Blobstream's smart contract events, the node most use a websocket connection, since an http one will not support subscriptions. This RPC url is different from the `"parent-chain.url" used in the node config, and is not necessary when running a full node.

To establish a WebSocket connection for your rollup to Arbitrum Sepolia, it's recommended to [find an RPC provider with WSS connections from Arbitrum's docs](https://docs.arbitrum.io/build-decentralized-apps/reference/node-providers). For this example, we will make an account on Alchemy. WebSocket (WSS) URLs which are essential for real-time data fetching and interaction with the Arbitrum Sepolia network. Follow these steps to set up your account and obtain a WSS URL using Alchemy:

1. Visit [Alchemy's website](https://www.alchemy.com/) and sign up for an account.
2. Once logged in, create a new app by selecting the Arbitrum network, specifically targeting the Arbitrum Sepolia testnet.
3. After creating your app, navigate to the "API key" section to find your WebSocket (WSS) URL.
4. Use this WSS URL in your nodeConfig.json under the celestia-cfg.eth-rpc object to ensure your node can establish a WebSocket connection to the Arbitrum Sepolia network and thus successfully subscribe to Blobstream events. 

Without a WSS connection, the Batch Poster won't be able to subscribe to Blobstream events, and thus will fallback to posting data to parent chain.
```

Reply:
https://github.com/celestiaorg/docs/pull/1490/commits/3a5694d2e1698f19c1d5a5e028ea7d93566f5ff0